bravado/swagger_model.py

```diff
@@ -24,22 +27,25 @@ class FileEventual(object):
     class FileResponse(object):
 
         def __init__(self, data):
-            self.data = data
+            self.text = data
+            self.headers = {}
 
         def json(self):
-            return self.data
+            return self.text
 
     def __init__(self, path):
         self.path = path
+        self.is_yaml = is_yaml(path)
 
     def get_path(self):
-        if not self.path.endswith('.json'):
+        if not self.path.endswith('.json') and not self.is_yaml:
             return self.path + '.json'
         return self.path
 
     def wait(self, timeout=None):
         with contextlib.closing(urllib.request.urlopen(self.get_path())) as fp:
-            return self.FileResponse(json.load(fp))
+            content = fp.read() if self.is_yaml else json.load(fp)
```

Comment:
For the yaml case, why is this setting `content` to the raw contents instead of `yaml.load(fp)`?  I actually think `content` would benefit from being renamed since in both cases, it is the Python object that represents the ingested json or yaml file.